motoman_driver/include/motoman_driver/simple_message/motoman_select_tool.h

```diff
@@ -0,0 +1,212 @@
+{
+namespace misc
+{
+
+/**
+ * \brief Class encapsulated select tool data. Motoman specific interface
+ * to select a specific tool file on the controller.
+ *
+ * The byte representation of a select tool command is as follows
+ * (in order lowest index to highest). The standard sizes are given,
+ * but can change based on type sizes:
+ *
+ *   member:             type                                      size
+ *   group_number        (industrial::shared_types::shared_int)    4  bytes
+ *   tool_number         (industrial::shared_types::shared_int)    4  bytes
+ *   sequence_number     (industrial::shared_types::shared_int)    4  bytes
+ *
+ * Note: sequence_number is marked as optional by MotoROS. It will be included
+ * in the SimpleMessage service request, but does not have to be set to any
+ * specific value. It can be used as a "tracking number" that will be "echoed
+ * back in the response message" (according to MotoROS source code comments).
+ *
+ * THIS CLASS IS NOT THREAD-SAFE
+ *
+ */
+
+class SelectTool : public industrial::simple_serialize::SimpleSerialize
+{
+public:
+  /**
+   * \brief Default constructor
+   *
+   * This method creates empty data.
+   *
+   */
+  SelectTool(void);
+
+  /**
+   * \brief Destructor
+   *
+   */
+  ~SelectTool(void);
+
+  /**
+   * \brief Initializes a empty select tool command
+   *
+   */
+  void init();
+
+  /**
+   * \brief Initializes a complete select tool command
+   *
+   * Note: sequence_number default value of 0.
+   */
+  void init(industrial::shared_types::shared_int group_number,
+    industrial::shared_types::shared_int tool_number,
+    industrial::shared_types::shared_int sequence_number = 0);
+
+  /**
+   * \brief Sets group number
+   *
+   * \param group_number Controller group the tool file is defined for
+   */
+  void setGroupNumber(industrial::shared_types::shared_int group_number)
+  {
+    this->group_number_ = group_number;
+  }
+
+  /**
+   * \brief Returns the group number
+   *
+   * \return group number
+   */
+  industrial::shared_types::shared_int getGroupNumber()
+  {
+    return this->group_number_;
+  }
+
+  /**
+   * \brief Sets tool number
+   *
+   * \param tool_number Tool file identifier
+   */
+  void setToolNumber(industrial::shared_types::shared_int tool_number)
+  {
+    this->tool_number_ = tool_number;
+  }
+
+  /**
+   * \brief Returns the tool file identifier
+   *
+   * \return tool number
+   */
+  industrial::shared_types::shared_int getToolNumber()
+  {
+    return this->tool_number_;
+  }
+
+  /**
+   * \brief Sets sequence number
+   *
+   * \param sequence_number Optional sequence number
+   */
+  void setSequenceNumber(industrial::shared_types::shared_int sequence_number)
+  {
+    this->sequence_number_ = sequence_number;
+  }
+
+  /**
+   * \brief Returns the optional sequence number
+   *
+   * \return sequence number
+   */
+  industrial::shared_types::shared_int getSequenceNumber()
+  {
+    return this->sequence_number_;
+  }
```

Comment:
Why not just expose these variables as public members instead of making getters and setters? It would cut out a lot of unnecessary code

Reply:
As I wrote earlier: this is all copy-paste and search-and-replace.

We could rewrite it all, but that would mean it's no longer a search-and-replace job.

As to why it was written like this: we'd have to ask the previous developers.

Reply:
There's nothing technically wrong here, so I'm okay with keeping it in the interest of merging proven and valuable functionality without adding too much additional effort